lib/annotate/annotate_models.rb

```diff
@@ -611,6 +611,15 @@ def get_loaded_model(model_path, file)
         $LOAD_PATH.map(&:to_s)
                   .select { |path| absolute_file.include?(path) }
                   .map { |path| absolute_file.sub(path, '').sub(/\.rb$/, '').sub(/^\//, '') }
+
+      # Handle Rails apps with collapsed model paths
+      model_paths = model_paths
+        .select do |model_path|
+          defined?(Rails) &&
+          Rails.autoloaders.main.collapse_dirs.any? &&
+          Rails.autoloaders.main.collapse_dirs.select { |path| path.match(model_path) }
+        end.map { |model_path| model_path.sub(/\/models/, '') }
```

Comment:
My concern here is simply removing `"/models"` from the path is a hardcoded approach. To me, it could be great if we could use `ActiveSupport::Inflector` or `Zeitwerk` to define the expected path, as Zeitwerk enforces the behaviour set using `Rails.autoloaders.main.collapse(...)`.